File: src/main/kotlin/no/nav/fo/forenkletdeploy/commits/GithubCommitProvider.java
package no.nav.fo.forenkletdeploy.commits;

import no.nav.fo.forenkletdeploy.domain.ApplicationConfig;
import no.nav.fo.forenkletdeploy.domain.Commit;
import org.springframework.stereotype.Component;

import javax.inject.Inject;
import java.util.List;

@Component
public class GithubCommitProvider {
    private final StashCommitProvider stashCommitProvider;

    @Inject
    public GithubCommitProvider(StashCommitProvider stashCommitProvider) {
        this.stashCommitProvider = stashCommitProvider;
    }

    public List<Commit> getCommitsForRelease(ApplicationConfig application, String fromTag, String toTag) {
        ApplicationConfig applicationStash = ApplicationConfig.builder()
                .gitUrl(String.format("ssh://git@stash.devillo.no:7999/fa/%s.git", application.getName()))
                .name(application.name)
                .build();

        return stashCommitProvider.getCommitsForRelease(applicationStash, fromTag, toTag);
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/commits/GithubCommitProvider.kt
package no.nav.fo.forenkletdeploy.commits

import no.nav.fo.forenkletdeploy.ApplicationConfig
import no.nav.fo.forenkletdeploy.Commit
import org.springframework.stereotype.Service
import javax.inject.Inject

@Service
class GithubCommitProvider @Inject constructor(val stashCommitProvider: IStashCommitProvider) {
    fun getCommitsForRelease(application: ApplicationConfig, fromTag: String, toTag: String): List<Commit> =
        stashCommitProvider.getCommitsForRelease(
                fromTag = fromTag,
                toTag = toTag,
                application = ApplicationConfig(
                        gitUrl = "ssh://git@stash.devillo.no:7999/fa/${application.name}.git",
                        name = application.name
                )
        )
}